code/game/machinery/computer/arcade/orion_event.dm

```diff
@@ -0,0 +1,510 @@
+
+//yep you guessed it back at it again with another datum singleton.
+/datum/orion_event
+	var/name = "this displays the events name"
+	var/text = "this displays a blurb about the event"
+	///pickweight to show up. will still be in the events pool if added to the events list but not RANDOM, only triggered.
+	var/weight = 0
+	///buttons to pick in response to the event. Don't worry, orion js will handle the rest
+	var/list/event_responses = list()
+	///default emag effect of events is to play an audible message and sound
+	var/emag_message = "<span class='userdanger'>Coders forgot to set this!</span>"
+	var/emag_sound = 'sound/effects/bamf.ogg'
+	///our game
+	var/obj/machinery/computer/arcade/orion_trail/game
+	///gaming skill level of the player
+	var/gamerSkillLevel = 0
+	///gaming skill of the player
+	var/gamerSkill = 0
+	///some other metric that makes it easier to do randoms with skill testing, god really 3 vars guys
+	var/gamerSkillRands = 0
+
+/datum/orion_event/New(_game)
+	. = ..()
+	game = _game
+
+/**
+ * What happens when this event is selected to trigger, sets vars. also can set some event pre-encounter randomization
+ *
+ * Arguments:
+ * * _gamerSkill: gaming skill of the player
+ * * _gamerSkillLevel: gaming skill level of the player
```

Comment:
More of the `src.var = var` pattern here, please. Documentation with underscore-prefixed names is weird, and shows off the implementation detail.